pkg/storage/url/urldownload/storage.go

```diff
@@ -0,0 +1,123 @@
+package urldownload
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"net/url"
+	"path/filepath"
+
+	"github.com/filecoin-project/bacalhau/pkg/config"
+	"github.com/filecoin-project/bacalhau/pkg/storage"
+	"github.com/filecoin-project/bacalhau/pkg/system"
+	"github.com/go-resty/resty/v2"
+	"github.com/rs/zerolog/log"
+	"go.opentelemetry.io/otel/trace"
+)
+
+// a storage driver runs the downloads content
+// from a public URL source and copies it to
+// a local directory in preparation for
+// a job to run - it will remove the folder/file once complete
+
+type StorageProvider struct {
+	LocalDir   string
+	HTTPClient *resty.Client
+}
+
+func NewStorageProvider(cm *system.CleanupManager) (*StorageProvider, error) {
+	// TODO: consolidate the various config inputs into one package otherwise they are scattered across the codebase
+	dir, err := ioutil.TempDir(config.GetStoragePath(), "bacalhau-url")
+	if err != nil {
+		return nil, err
+	}
+
+	client := resty.New()
+	// Setting output directory path, If directory not exists then resty creates one
+	client.SetOutputDirectory(dir)
+
+	storageHandler := &StorageProvider{
+		HTTPClient: client,
+		LocalDir:   dir,
+	}
+
+	log.Debug().Msgf("URL download driver created with output dir: %s", dir)
+	return storageHandler, nil
+}
+
+func (sp *StorageProvider) IsInstalled(ctx context.Context) (bool, error) {
+	_, span := newSpan(ctx, "IsInstalled")
+	defer span.End()
+	return true, nil
+}
+
+func (sp *StorageProvider) HasStorageLocally(ctx context.Context, volume storage.StorageSpec) (bool, error) {
+	_, span := newSpan(ctx, "HasStorageLocally")
+	defer span.End()
+	return false, nil
+}
+
+// Could do a HEAD request and check Content-Length, but in some cases that's not guaranteed to be the real end file size
+func (sp *StorageProvider) GetVolumeSize(ctx context.Context, volume storage.StorageSpec) (uint64, error) {
+	return 0, nil
+}
+
+func (sp *StorageProvider) PrepareStorage(ctx context.Context, storageSpec storage.StorageSpec) (storage.StorageVolume, error) {
+	_, span := newSpan(ctx, "PrepareStorage")
+	defer span.End()
+
+	_, err := IsURLSupported(storageSpec.URL)
+	if err != nil {
+		return storage.StorageVolume{}, err
+	}
+
+	outputPath, err := ioutil.TempDir(sp.LocalDir, "*")
+	if err != nil {
+		return storage.StorageVolume{}, err
+	}
+
+	sp.HTTPClient.SetTimeout(config.GetDownloadURLRequestTimeout())
+	_, err = sp.HTTPClient.R().
+		SetOutput(outputPath + "/file").
+		Get(storageSpec.URL)
+	if err != nil {
+		return storage.StorageVolume{}, err
+	}
+
+	volume := storage.StorageVolume{
+		Type:   storage.StorageVolumeConnectorBind,
+		Source: outputPath + "/file",
+		Target: storageSpec.Path,
+	}
+
+	return volume, nil
+}
+
+// nolint:lll // Exception to the long rule
+func (sp *StorageProvider) CleanupStorage(ctx context.Context, storageSpec storage.StorageSpec, volume storage.StorageVolume) error {
```

Comment:
Just a general code style comment - I always tend toward the following style to get around long line length complaints from the linter:

```go
func (sp *StorageProvider) CleanupStorage(
  ctx context.Context,
  storageSpec storage.StorageSpec,
  volume storage.StorageVolume,
) error {
```

Reply:
Fixed thanks!